### ISSUE TITLE
.NET Core 8.0 WinForms TreeView null reference exception 'owningTreeView'

### ISSUE THREAD

**Issue by dyerjohn** [dotnet/winforms#10876]
### .NET version

.NET 8 Core

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

This code has been working since .NET Framework 3.5.
It's been migrated to Framework 4.x
Then .NET Core 3.1
And lastly .NET Core 6.0
It's now in .NET Core 8.0 that there is a problem.

### Issue description

I have code that has been migrated from .NET Framework over to .NET Core 3.1 then 6.0 and now I’m migrating to 8.0. It’s worked fine in all versions up to 8.0 where I’m now getting a null reference exception internal to the Core code.

Setting breakpoints will cause the error to become less likely to happen which leads me to suspect that this is some weird Windows messaging related issue where a slight change in the code execution time allows messages to process slightly different. I’m posting now in the hope that this is a known issue or with a known solution. I will be continuing to work on debugging further to narrow down what’s going on.

The exception:
System.ArgumentNullException
HResult=0x80004003
Message=Value cannot be null. (Parameter ‘owningTreeView’)
Source=System.Private.CoreLib
StackTrace:
at System.ArgumentNullException.Throw(String paramName) in /_/src/libraries/System.Private.CoreLib/src/System/ArgumentNullException.cs:line 87

Top call stack entries:
System.Private.CoreLib.dll! System.ArgumentNullException.Throw(string paramName) Line 87 C#
System.Windows.Forms.Primitives.dll! System.Windows.Forms.ArgumentValidation.OrThrowIfNull<System.__Canon>(System.__Canon argument, string paramName) Line 12 C#
System.Windows.Forms.dll! System.Windows.Forms.TreeNode.TreeNodeAccessibleObject.TreeNodeAccessibleObject(System.Windows.Forms.TreeNode owningTreeNode, System.Windows.Forms.TreeView owningTreeView) Line 19 C#
System.Windows.Forms.dll! System.Windows.Forms.TreeNode.AccessibilityObject.get() Line 1172 C#
System.Windows.Forms.dll! System.Windows.Forms.TreeView.OnAfterSelect(System.Windows.Forms.TreeViewEventArgs e) Line 2330 C#
System.Windows.Forms.dll! System.Windows.Forms.TreeView.TvnSelected(Interop.ComCtl32.NMTREEVIEW* nmtv) Line 2663 C#
System.Windows.Forms.dll! System.Windows.Forms.TreeView.WmNotify(ref System.Windows.Forms.Message m) Line 3112 C#
System.Windows.Forms.dll! System.Windows.Forms.TreeView.WndProc(ref System.Windows.Forms.Message m) Line 3343 C#
System.Windows.Forms.dll! System.Windows.Forms.NativeWindow.Callback(Windows.Win32.Foundation.HWND hWnd, Windows.Win32.MessageId msg, Windows.Win32.Foundation.WPARAM wparam, Windows.Win32.Foundation.LPARAM lparam) Line 372 C#

### Steps to reproduce

This only seems to happen when clicking on TreeView items that have sub items.  
I'm continuing to investigate.

**Comment by Tanya-Solyanik** [comment]
The related code was introduced in https://github.com/dotnet/winforms/pull/6432

@dyerjohn  - no, this is not known. Thank you for raising this issue!

What else besides selecting nodes with subnodes is happening in your program? Are you moving subnodes between trees or nodes? Is the tree dynamically created or modified, are you closing the containing form?

**Comment by Tanya-Solyanik** [comment]
Thank you, @dyerjohn !

>it's a tree that gets build on FormLoad

This is what I would expect would need to happen to select an unparented node, but I had not hit this condition yet.

>you can drag subnodes around

This is a good hint! I tried adding and removing subtrees, not dragging.

Is there anything specific that the user does before clickong on a node with subnodes? I'm concerned that my fix might not be sufficient. Would you be interested to test it?

**Comment by dyerjohn** [comment]
@Tanya-Solyanik  Sure I can test the fix.  How do I get the updated DLL ?

**Comment by Tanya-Solyanik** [comment]
@dyerjohn  - in a couple of days this fix will get into the NET9 sdk, latest version.

**Comment by dyerjohn** [comment]
@Tanya-Solyanik Is there a way to use the .NET 9 DLL in the .NET 8 system?  It's a bit of effort to re-work the csproj files.
When will the fix appear in .NET 8?

**Comment by Tanya-Solyanik** [comment]
@dyerjohn  - it might be easier to build System.Windows.Forms.dll from this PR - https://github.com/dotnet/winforms/pull/10910

**Comment by dyerjohn** [comment]
@Tanya-Solyanik So I'm not sure how to proceed from here.  I tried to download the source for the 8.0 and merge in two changed forms files but I'm getting other errors when I try and run the program.  Apparently I need to build a lot more than just WinForms bits.  I didn't really want to retarget the system to 9.0 for a test but it's looking like that is what I'll need to do.

To be clear, once this is tested on my side, will this change make its way into the 8.0 builds?  The whole reason for migrating from 6.0 to 8.0 was to get out code onto the next LTS build of .NET Core.  9.0 is another short term platform and not that useful to our project.

**Comment by Tanya-Solyanik** [comment]
@dyerjohn  - We will include this change into the next update for NET8, please install it when it's out.

**Comment by dyerjohn** [comment]
@Tanya-Solyanik Excellent!

**Comment by dyerjohn** [comment]
@Tanya-Solyanik I see that 8.0.3 is "kinda" out.  Checking the release page:
https://github.com/dotnet/core/blob/main/release-notes/8.0/8.0.3/8.0.3.md

A lot of the links reference back to 8.0.202.  The header text says: The .NET 8.0.3 and .**NET SDK 8.0.202** releases are available for download.  Is this a mistake?  If not, what's the reason of mixing two different versions on the same release page?

**Comment by mschmidt98** [comment]
Hi,

we have a similar problem when changing the selected node, which triggers some actions like reloading and filtering data in our program. 
For testing, if the 8.0.3 did the trick, I built a version disabling roll-foward and specifing runtime version to 8.0.3.

The crash still occurs, the current stacktrace is

> System.ArgumentNullException: Value cannot be null. (Parameter 'owningTreeView')
   at System.ArgumentNullException.Throw(String paramName)
   at System.Windows.Forms.TreeNode.TreeNodeAccessibleObject..ctor(TreeNode owningTreeNode, TreeView owningTreeView)
   at System.Windows.Forms.TreeNode.get_AccessibilityObject()
   at System.Windows.Forms.TreeView.OnAfterSelect(TreeViewEventArgs e)
   at System.Windows.Forms.TreeView.TvnSelected(NMTREEVIEW* nmtv)
   at System.Windows.Forms.TreeView.WmNotify(Message& m)
   at System.Windows.Forms.TreeView.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(HWND hWnd, MessageId msg, WPARAM wparam, LPARAM lparam)

On my win 11 development machine I was not able to reproduce this, as well as a fresh VM of Server 2022 Desktop. Only on our terminal server, which is running Server 2022 20348 this is reproducable for me. Is it possible that there are some accessibility settings of the OS interfering with this?

**Comment by Tanya-Solyanik** [comment]
Thanks, @mschmidt98, please update this item with your results.
